cabal-install/Distribution/Solver/Modular/Log.hs

```diff
@@ -40,25 +40,25 @@ logToProgress mbj l = let
     -- and ignores repeated backjumps. If proc reaches the backjump limit, it truncates
     -- the 'Progress' and ends it with the last conflict set. Otherwise, it leaves the
     -- original success result or replaces the original failure with 'Nothing'.
-    proc :: Maybe Int -> Progress Message a b -> Progress Message (Maybe (ConflictSet QPN)) b
+    proc :: Maybe Int -> Log Message b -> Progress Message (Bool, ConflictSet QPN) b
     proc _        (Done x)                          = Done x
-    proc _        (Fail _)                          = Fail Nothing
+    proc _        (Fail (cs, _cm))                  = Fail (True, cs)
     proc mbj'     (Step x@(Failure cs Backjump) xs@(Step Leave (Step (Failure cs' Backjump) _)))
       | cs == cs'                                   = Step x (proc mbj'           xs) -- repeated backjumps count as one
-    proc (Just 0) (Step   (Failure cs Backjump)  _) = Fail (Just cs)
+    proc (Just 0) (Step   (Failure cs Backjump)  _) = Fail (False, cs)
     proc (Just n) (Step x@(Failure _  Backjump) xs) = Step x (proc (Just (n - 1)) xs)
     proc mbj'     (Step x                       xs) = Step x (proc mbj'           xs)
 
     -- Sets the conflict set from the first backjump as the final error, and records
     -- whether the search was exhaustive.
-    useFirstError :: Progress Message (Maybe (ConflictSet QPN)) b
-                  -> Progress Message (Bool, Maybe (ConflictSet QPN)) b
+    useFirstError :: Progress Message (Bool, ConflictSet QPN) b
+                  -> Progress Message (Bool, ConflictSet QPN) b
     useFirstError = replace Nothing
       where
         replace _       (Done x)                          = Done x
-        replace cs'     (Fail cs)                         = -- 'Nothing' means backjump limit not reached.
+        replace cs'     (Fail (exh, cs))                  = -- 'Nothing' means backjump limit not reached.
                                                             -- Prefer first error over later error.
```

Comment:
This comment needs updating (something about doing something specifically when you exhaust search space?)